sycl/include/sycl/nd_item.hpp

```diff
@@ -38,6 +41,502 @@ namespace ext::oneapi::experimental {
 template <int Dimensions> class root_group;
 }
 
+#if __INTEL_PREVIEW_BREAKING_CHANGES
+/// Identifies an instance of the function object executing at each point in an
+/// nd_range.
+///
+/// \ingroup sycl_api
+template <int Dimensions = 1> class nd_item {
+public:
+  static constexpr int dimensions = Dimensions;
+
+  id<Dimensions> get_global_id() const {
+#ifdef __SYCL_DEVICE_ONLY__
+    return __spirv::initGlobalInvocationId<Dimensions, id<Dimensions>>();
+#else
+    throw sycl::exception(sycl::make_error_code(sycl::errc::invalid),
+                          "nd_item methods can't be invoked on the host");
+    return {};
+#endif
+  }
+
+  size_t __SYCL_ALWAYS_INLINE get_global_id(int Dimension) const {
+    size_t Id = get_global_id()[Dimension];
+    __SYCL_ASSUME_INT(Id);
+    return Id;
+  }
+
+  size_t __SYCL_ALWAYS_INLINE get_global_linear_id() const {
```

Comment:
I wonder whether there is a potential performance impact here for code which calls `get_global_linear_id()` quite liberally, how does the code generated for this implementation compare with the original implementation?

Reply:
They match. 
As I removed the `item` member, I just inlined `item::get_linear_id` implementation. 

https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/detail/item_base.hpp#L33-L42

I don't expect any performance changes in this regard.

You might find [this discussion](https://github.com/intel/llvm/pull/12236#discussion_r1434590193) interesting. I tried to use SPIR-V intrinsic to get liner ID, but it broke some precommit tests and I wasn't sure how to handle offsets on NVIDIA and AMD GPUs.